### ISSUE TITLE
HOW TO BUILD ON IOS!!!

### ISSUE THREAD

**Issue by thieuhuyenuit88** [Rakashazi/emu-ex-plus-alpha#89]
someone has a clearly tutorial for ios build?

**Comment by NE1WeKnow** [comment]
I am working through this myself and struggling.  

There is already another open issue discussing build instructions.
[https://github.com/Rakashazi/emu-ex-plus-alpha/issues/85](url)

Also, look at README files for dependencies on the emu you wish to build
[https://github.com/Rakashazi/emu-ex-plus-alpha/blob/master/imagine/doc/INSTALL](url)
[https://github.com/Rakashazi/emu-ex-plus-alpha/blob/master/EmuFramework/doc/INSTALL](url)

**Comment by thededman** [comment]
I have worked on this project on and off a long time:
https://github.com/thededman/emu-ex-plus-alpha/wiki

NOTE issues is the current build WILL NOT work on iOS 9/10.x, it will compile and load but will not browse to the ROM files in the app. The program is escaping the sandbox (documents) folder.
If you try your hand at my instruction used this version: https://github.com/Rakashazi/emu-ex-plus-alpha/tree/58ccd51c80881bef6daae925ee796eeb90e70853

This error was noted: https://github.com/Rakashazi/emu-ex-plus-alpha/issues/73

I will try and update my fork WIKI page in a month, I'm tried up with completing CISSP.
Rakashazi is busy as well so not have answers quickly. I hope my notes help. Good Luck.

**Comment by thieuhuyenuit88** [comment]
@thededman thank you so much
I follow your note and when i build the Emulator (NES.EMU) i got a error like this:

CFLAGS -I/opt/local/lib/pkgconfig/../../include -std=gnu++14 -fno-rtti -fno-exceptions -fno-threadsafe-statics -stdlib=libc++  -pipe -fvisibility=hidden -O2 -fomit-frame-pointer -fno-stack-protector  -flto -mdynamic-no-pic -Wall -Wextra -Werror=return-type -Wno-comment -Wno-missing-field-initializers -Wno-unused -Wno-unused-parameter  -Wdisabled-optimization -Wno-attributes -Wno-missing-braces -Woverloaded-virtual -o build/ios-release-armv7/obj/main/Main.o
clang: clang: error: no such file or directory: 'CFLAGS'
error: no such file or directory: 'CFLAGS'
clang: error: no such file or directory: 'CFLAGS'
clang: error: no such file or directory: 'CFLAGS'

i can guess my problem is PKG_CONFIG_PATH but i don't know how to set this path. any help? thanks.

**Comment by NE1WeKnow** [comment]
Readme file here suggests that PKG_CONFIG_PATH is not really a requirement:
[https://github.com/Rakashazi/emu-ex-plus-alpha/blob/master/imagine/doc/INSTALL](url)

But anyway, this article has instructions for setting path:
[http://hathaway.cc/post/69201163472/how-to-edit-your-path-environment-variables-on-mac](url)

**Comment by NE1WeKnow** [comment]
If you have success with NES.EMU, maybe you could build unsigned C64.EMU IPA for me as I am not getting anywhere... :-(

**Comment by thieuhuyenuit88** [comment]
i have changed environment variable PKG_CONFIG_PATH but it still this error:
..........
CFLAGS -I/Users/NGUYENDANTRUONG/imagine-sdk/ios-arm64/lib/pkgconfig/../../include/minizip -I/opt/local/lib/pkgconfig/../../include
...........
error: no such file or directory: 'CFLAGS'
i have no idea to fix this error. Please help me :(

**Comment by NE1WeKnow** [comment]
Is DEB file of use to you?  If so, I can help you out.  Reach me at ne1weknow (at) gmail (dot) com

**Comment by thieuhuyenuit88** [comment]
@NE1WeKnow i don't know what is DEB file? Can you help me? thanks!

**Comment by thededman** [comment]
YES! the DEB is the emulator program. It will not work as is, so you need an Empty Xcode Frame.
I covered this in my fork: look at instruction number 6: https://github.com/thededman/emu-ex-plus-alpha
Sorry no screenshots, I've been busy with career and kids! I will do my best to help as I did others in this project!

**Comment by thededman** [comment]
I have updated the WIKI on fork. I need to get this out of onenote and into git, hope this helps.

https://github.com/thededman/emu-ex-plus-alpha/wiki/emu-ex-plus-alpha-How-to-build-(MacOSX--XCode-Only)

**Comment by NE1WeKnow** [comment]
That's good info you have provided - I think there's not much which has changed except screenshots added and highlighting of commands.  

I do agree somewhat that there is fun to be had in compiling the project yourself.  It was fun for other projects when I created my VM, installed Xcode and followed tutorials and produced signed IPAs at the end which were pushed to my device and worked.  But then the 7-day expiry for free developer accounts came in and this project has to be the toughest one yet to get a working build for due to required dependencies.

**Comment by ck16** [comment]
@thededman Thanks for your guide.  I managed to get it all compiled and noticed an omission in your guide.  MacPorts also requires the automake package in order to compile the dependancies.

I don't think the MacPorts installation help was in your main compiling guide, but I remember reading something you wrote elsewhere.

I also noticed the skeleton Xcode project for GBC.emu is missing.

Thanks heaps for all your hard work!

Chris
